c/loops/terminator_02_true-unreach-call_true-termination.c

```diff
@@ -12,27 +12,28 @@ _Bool __VERIFIER_nondet_bool();
 
 int main()
 {
-    int x=__VERIFIER_nondet_int();
-    int y=__VERIFIER_nondet_int();
-    int z=__VERIFIER_nondet_int();
-    if (!(x<100)) return 0;
-    if (!(x>-100)) return 0;
-    if (!(z<100)) return 0;
-    if (!(z>-100)) return 0;
-    while(x<100 && 100<z) 
-    {
-        _Bool tmp=__VERIFIER_nondet_bool();
-        if (tmp) {
-            x++;
-        } else {
-            x--;
-            z--;
-        }
-    }                       
-
-    __VERIFIER_assert(x>=100 || z<=100);
-
-    return 0;
+  int x=__VERIFIER_nondet_int();
+  int z=__VERIFIER_nondet_int();
+
+  __VERIFIER_assume(x>-100);
+  __VERIFIER_assume(x<200);
+  __VERIFIER_asume(z>100);
+  __VERIFIER_assume(z<200);
```

Comment:
There was #352 by @jhensel which in c856da1fc81e227df30bcc0bac1430d9531fba58 explicitly replaced the `__VERIFIER_assume` with `if (...) return 0;`. Apparently this was done for termiination, but I don't know why.

@dbeyer @jhensel @tautschnig @kfriedberger What was the reason for this? Is `__VERIFIER_assume` really a problem for termination? Why is this then a problem only in the loops category?

Reply:
These benchmarks were labelled as terminating although __VERIFIER_assume is defined as

void __VERIFIER_assume(int expression) { if (!expression) { LOOP: goto LOOP; }; return; }

When I asked to relabel them, someone had the idea to replace the assume such that the label is correct.

Reply:
I believe #348 should provide the insights on `__VERIFIER_assume` vs `if (...) return 0;`.

Reply:
@jhensel I wanted to let you know that there is some discussion about whether `__VERIFIER_assume` causes non-termination here: https://github.com/sosy-lab/sv-benchmarks/issues/504#issuecomment-344938369
It seems that other people are of the opposite opinion and have marked tasks accordingly in the past (i.e., with `__VERIFIER_assume` being interpreted similarly to `abort`).
You might want to join this discussion.